### PR TITLE
RHEL 8 Compatibility

### PR DESCRIPTION
**Pull request by nv-kmcgill53**
- Grab the distribution family from the ID_LIKE variable in the /etc/os-release file.
- Add explicit libraries needed during linking

### CODE REVIEW

File: src/test/CMakeLists.txt [hunk 1]
```diff
@@ -680,8 +681,8 @@ if(${TRITON_ENABLE_GPU})
       GTest::gtest_main
       GTest::gmock
       protobuf::libprotobuf
-      ${CNMEM_LIBRARY}
       CUDA::cudart
+      ${CNMEM_LIBRARY}
```
Comment (fpetrini15) [inline src/test/CMakeLists.txt:685]: Was this move functional or stylistic?

**Review by fpetrini15 — APPROVED** [review]
LGTM!

**Review by fpetrini15 — APPROVED** [review]

File: src/test/CMakeLists.txt [hunk 2]
```diff
@@ -672,6 +672,7 @@ if(${TRITON_ENABLE_GPU})
   target_link_libraries(
     input_byte_size_test
     PRIVATE
+      dl
```
[the remark this answers is not in the document]
Comment (fpetrini15) [reply src/test/CMakeLists.txt:675]: I believe it stands for Dynamically Loaded (DL) Libraries: https://tldp.org/HOWTO/Program-Library-HOWTO/dl-libraries.html#:~:text=Dynamically%20loaded%20(DL)%20libraries%20are,the%20plugin%20until%20it's%20needed.
It gives us access to `dlopen`, `dlerror`, etc.

[the remark this answers is not in the document]
Comment (nv-kmcgill53) [reply src/test/CMakeLists.txt:675]: What @fpetrini15 said. For some reason this library needs to be explicitly listed in the rhel-like distros but not in the debian distro.

**Review by fpetrini15 — APPROVED** [review]

**Pull request merged**